openedx/core/djangoapps/content_libraries/tests/test_runtime.py

```diff
@@ -33,37 +36,35 @@ class ContentLibraryContentTestMixin:
     """
     Mixin for content library tests that creates two students and a library.
     """
-    @classmethod
-    def setUpClass(cls):
-        super().setUpClass()
+    def setUp(self):
+        super().setUp()
         # Create a couple students that the tests can use
-        cls.student_a = UserFactory.create(username="Alice", email="alice@example.com", password="edx")
-        cls.student_b = UserFactory.create(username="Bob", email="bob@example.com", password="edx")
+        self.student_a = UserFactory.create(username="Alice", email="alice@example.com", password="edx")
+        self.student_b = UserFactory.create(username="Bob", email="bob@example.com", password="edx")
+
         # Create a collection using Blockstore API directly only because there
         # is not yet any Studio REST API for doing so:
-        cls.collection = blockstore_api.create_collection("Content Library Test Collection")
+        self.collection = blockstore_api.create_collection("Content Library Test Collection")
         # Create an organization
-        cls.organization = Organization.objects.create(
+        self.organization = Organization.objects.create(
             name="Content Libraries Tachyon Exploration & Survey Team",
             short_name="CL-TEST",
         )
-        cls.library = library_api.create_library(
-            collection_uuid=cls.collection.uuid,
+        _, slug = self.id().rsplit('.', 1)
+        self.library = library_api.create_library(
+            collection_uuid=self.collection.uuid,
             library_type=COMPLEX,
-            org=cls.organization,
-            slug=cls.__name__,
-            title=(cls.__name__ + " Test Lib"),
+            org=self.organization,
+            slug=slugify(slug),
+            title=(f"{slug} Test Lib"),
             description="",
             allow_public_learning=True,
             allow_public_read=False,
             library_license=ALL_RIGHTS_RESERVED,
         )
 
 
-@requires_blockstore
-# EphemeralKeyValueStore requires a working cache, and the default test cache doesn't work:
-@override_settings(XBLOCK_RUNTIME_V2_EPHEMERAL_DATA_CACHE='blockstore')
-class ContentLibraryRuntimeTest(ContentLibraryContentTestMixin, TestCase):
+class ContentLibraryRuntimeTestMixin(ContentLibraryContentTestMixin):
```

Comment:
Hmm, most of the tests in this file are being skipped? Why is that the case?

```
openedx/core/djangoapps/content_libraries/tests/test_runtime.py sssssss....sssssssssssss
```

Reply:
Most of these tests are [skipped in the CMS](https://github.com/openedx/edx-platform/blob/d368dc03aaf46b468799cc18f21535456237b51c/openedx/core/djangoapps/content_libraries/tests/test_runtime.py#L73) or [skipped in the LMS](https://github.com/openedx/edx-platform/blob/d368dc03aaf46b468799cc18f21535456237b51c/openedx/core/djangoapps/content_libraries/tests/test_runtime.py#L206), so you have to run them in both environments in order to see them pass (like the CI does).

Half the tests are also [skipped if you're not running the blockstore service](https://github.com/openedx/edx-platform/blob/d368dc03aaf46b468799cc18f21535456237b51c/openedx/core/djangoapps/content_libraries/tests/test_runtime.py#L184), or if you're [running the blockstore app](https://github.com/openedx/edx-platform/blob/d368dc03aaf46b468799cc18f21535456237b51c/openedx/core/djangoapps/content_libraries/tests/test_runtime.py#L193).

So you have to run each permutation to hit them all:

```bash
# CMS
root@studio:/edx/app/edxapp/edx-platform# pytest  --ds cms.envs.test openedx/core/djangoapps/content_libraries/tests/test_runtime.py
root@studio:/edx/app/edxapp/edx-platform# EDXAPP_RUN_BLOCKSTORE_TESTS=1 pytest  --ds cms.envs.test openedx/core/djangoapps/content_libraries/tests/test_runtime.py

# LMS
root@studio:/edx/app/edxapp/edx-platform# pytest  --ds lms.envs.test openedx/core/djangoapps/content_libraries/tests/test_runtime.py
root@studio:/edx/app/edxapp/edx-platform# EDXAPP_RUN_BLOCKSTORE_TESTS=1 pytest  --ds lms.envs.test openedx/core/djangoapps/content_libraries/tests/test_runtime.py
```

Reply:
Got it. All of them worked in my testing as well.